packages/button/index.tsx

```diff
@@ -0,0 +1,99 @@
+// The MIT License
+//
+// Copyright (c) 2018 Google, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import * as React from 'react';
+import * as classnames from 'classnames';
+// @ts-ignore
+import withRipple from '@material/react-ripple';
+
+export interface ButtonProps<T> {
+  raised?: boolean,
+  unelevated?: boolean,
+  outlined?: boolean,
+  dense?: boolean,
+  disabled?: boolean,
+  unbounded?: boolean,
+  initRipple?: (surface: T) => void,
+  className?: string,
+  icon?: React.ReactElement<React.HTMLProps<HTMLOrSVGElement>>,
+  href?: string
+}
+
+type AnchorElementProps = React.AnchorHTMLAttributes<HTMLAnchorElement> & ButtonProps<HTMLAnchorElement>;
+type ButtonElementProps = React.ButtonHTMLAttributes<HTMLButtonElement> & ButtonProps<HTMLButtonElement>;
+
+export const Button = <T extends {}>(
```

Comment:
Would it be fair to make using `export function` as the syntax part of a conversion to TSX for all of your components?  That would allow you to make this a bit easier to read now without sweating the impact on consistency.

Reply:
To change all the FunctionalComponents to use `function`? I also heard from Amr that using `function` is the preferred way. However there are about 22 components and a lot of `const` being used, so I think I'd prefer to push this one off til later if that's ok.